### PR TITLE
Flush all the queue when there are more than 50 events queued

### PR DESCRIPTION
**Pull request by patedit**

### CODE REVIEW

File: src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java [hunk 1]
```diff
@@ -392,9 +392,13 @@ private void sendAllData(MPDbAdapter dbAdapter) {
 
             private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] urls) {
                 final RemoteService poster = getPoster();
-                final String[] eventsData = dbAdapter.generateDataString(table);
+                String[] eventsData = dbAdapter.generateDataString(table);
+                Integer queueCount = 0;
+                if (queueCount != null) {
```
Comment (yinfeiru) [inline src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java]: Why this check?

Comment (yinfeiru) [inline src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java]: Should you check `eventsData`?

File: src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java [hunk 2]
```diff
@@ -459,6 +463,12 @@ private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] u
                         sendEmptyMessageDelayed(FLUSH_QUEUE, mTrackEngageRetryAfter);
                         mFailedRetries++;
                         logAboutMessageToMixpanel("Retrying this batch of events in " + mTrackEngageRetryAfter + " ms");
+                        break;
```
Comment (yinfeiru) [inline src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java]: I'm not certain but do we want to break on `OutOfMemoryError`?

File: src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java [hunk 3]
```diff
@@ -459,6 +463,12 @@ private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] u
...
+                    }
+
+                    eventsData = dbAdapter.generateDataString(table);
+                    if (eventsData != null) {
+                        queueCount = Integer.valueOf(eventsData[2]);
```
Comment (yinfeiru) [inline src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java]: I know it's not your fault but can we give `2` a meaningful name? Everything single time I look at this code I have to figure out what `generateDataString` does.

Comment (yinfeiru) [inline src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java]: (we use magic index numbers all over the place in this file eg. https://github.com/mixpanel/mixpanel-android/pull/336/files#diff-7c00d8762ab328cad50eba1fecb26bc5R402    not very readable)

File: src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java [hunk 4]
```diff
@@ -392,9 +392,13 @@ private void sendAllData(MPDbAdapter dbAdapter) {
 
             private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] urls) {
                 final RemoteService poster = getPoster();
-                final String[] eventsData = dbAdapter.generateDataString(table);
+                String[] eventsData = dbAdapter.generateDataString(table);
+                Integer queueCount = 0;
+                if (queueCount != null) {
+                    queueCount = Integer.valueOf(eventsData[2]);
```
Comment (yinfeiru) [inline src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java]: what if `eventsData[2]` is null

File: src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java (hunk 3 above)
Comment (yinfeiru) [inline src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java]: also, please add `eventsData[2] != null` check

File: src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java (hunk 1 above)
Comment (patedit) [inline src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java]: My mistake. I was doing something else before with `queueCount`.

File: src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java (hunk 3 above)
Comment (patedit) [inline src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java]: `eventsData[2]` can't be `null` if `eventsData` is not `null`.

**Pull request merged**